### ISSUE TITLE
Styles Player settingsWindow override current window flags

### ISSUE THREAD

**Issue by Jean-Daniel** [google/ExoPlayer#10029]
ExoPlayer: 2.17.0
Android: at least 7.1 but probably other versions too.

---

I have a full screen activity that is configured to hide status and navigation bar like this:

```kotlin
override fun onStart() {
        super.onStart()

        // should be perform on each start
        val ctrl = ViewCompat.getWindowInsetsController(binding.root) ?: return
        ctrl.systemBarsBehavior = WindowInsetsControllerCompat.BEHAVIOR_SHOW_TRANSIENT_BARS_BY_SWIPE
        ctrl.hide(WindowInsetsCompat.Type.systemBars())
}
```

This activity contains a StyledPlayerView using default settings and `useController` and `controllerAutoShow` both set to `true`

When the Player setting button is clicked, it open the `settingsWindow` that does not replicate the parent window flags, and so make the status bar and navigation bar visible.

This is not something expected for a popup window, as from the user point of view, the main window remain the player window.

And the most annoying part is that when the `settingsWindow` is closed, Android does not hide the status bar and navigation bar automatically. Which means that is should be done programmatically.

But there is no API in ExoPlayer to monitor the settingsWindow visibility.

Expected behavior: That opening the settingsWindow does not change the status bar and navigation bar visibility.
Or at least, an easy way to workaround that issue by having a `settingsWindow` visibility listener.

**Comment by ojw28** [comment]
I agree this is a bug, but it's unclear to me how to fix this. If you have any ideas, please feel free to send a pull request. Thanks!

**Comment by ojw28** [comment]
It looks like this is quite difficult to fix, and that it's a broader platform issue with UI components that are displayed using a different window. There's a long-open issue about the problem [here](https://issuetracker.google.com/issues/156065476), which has unfortunately not received much attention.

I did find one way of making it work, but it would rely on the application layer using the (now-deprecated) `View.setSystemUiVisibility` method to set the immersive mode, rather than `WindowInsetsControllerCompat` or `WindowInsetsController`, so it's not a great solution. I'm following up internally to see if we can do something better.

**Comment by Jean-Daniel** [comment]
I did at least found out that I can easily workaround that issue by moving my "hide system bar" code into `onWindowFocusChanged(hasFocus: Boolean)` method.

The navigation still became visible while the popup is open, but it is automatically hidden when the popup is dismissed.

**Comment by ojw28** [comment]
Just to check something: In cases where you're seeing the navigation remain visible _after_ the popup was dismissed (unless you do something proactive such as in your comment above), was that only on certain API levels? Specifically, does it only occur on level API level 29 and earlier? I _think_ the navigation probably does automatically hide itself again on API level 30 and above.

The implication of this is that if you need the workaround of listening to the window focus change to re-hide the navigation, you should probably target that piece of code to only run on API level 29 and earlier. Then it'll be clear you can remove it when, far in the future, your app's minimum API level is increased to 30!

**Comment by Jean-Daniel** [comment]
You're right. I did just test with a API 30 device, and it properly hide back the system bars when the setting window is closed.
